src/template_utils.rs

```diff
@@ -13,7 +13,41 @@ pub use askama_escape::escape;
 
 pub static CACHE_NAME: &str = env!("CACHE_ID");
 
-pub type BaseContext<'a> = &'a (&'a Connection, &'a Catalog, Option<User>);
+pub type BaseContext<'a> = &'a (
+    &'a Connection,
+    &'a Catalog,
+    Option<User>,
+    Option<(String, String)>,
+);
+
+pub trait IntoContext {
```

Comment:
why not implementing `From<PlumeRocket>` for Context? It will auto implement `Into<Context>` for PlumeRocket, and you should be able to use `into()` with no need to import this trait everywhere

Reply:
Because Rust is complaining about the fact that all the parameters of the trait comes from another crate, and that it could cause conflicts, while I don't think it can? :woman_shrugging: But yeah, I tried it, and it doesn't work.

Reply:
ho, it's because BaseContext is not a true struct but a typedef to a tuple of basic types (in theory it could cause conflict, because the exact same trait can be implemented in plume-model, thus creating a conflict). This mean we can either convert BaseContext to be a true type, or implement Into in plume-models. I've tried the first one, but it requires some small but ugly changes so I'm not sure it would be better (you can take a look at the patch [here](https://framadrop.org/r/aXcBuKkLEW#rzbPsNSdq+XCFboTnRLdY8Qt3RNqqUYGgrnA8bfUoQQ=))